modules/speech-to-text-impl/src/main/java/org/opencastproject/speechtotext/impl/engine/WhisperCppEngine.java

```diff
@@ -0,0 +1,417 @@
+
+  /**
+   * {@inheritDoc}
+   *
+   * @see org.opencastproject.speechtotext.api.SpeechToTextEngine#generateSubtitlesFile(File, File, String, Boolean)
+   */
+
+  @Override
+  public Map<String, Object> generateSubtitlesFile(File mediaFile,
+      File preparedOutputFile, String language, Boolean translate)
+          throws SpeechToTextEngineException {
+
+    if (!FilenameUtils.isExtension(mediaFile.getPath(), "wav")) {
+      throw new SpeechToTextEngineException("WhisperC++ currently doesn't support any media extension other than wav");
+    }
+
+    String[] baseCommands = { whispercppExecutable,
+        mediaFile.getAbsolutePath(),
+        "--model", whispercppModel,
+        "-ovtt",
+        "-oj",
+        "--output-file", preparedOutputFile.getAbsolutePath().replaceFirst("[.][^.]+$", "")};
+
+    List<String> command = new ArrayList<>(Arrays.asList(baseCommands));
+
+    if (whispercppBeamSize.isSome()) {
+      command.add("-bs");
+      command.add(Integer.toString(whispercppBeamSize.get()));
+    }
+    if (whispercppMaxLength.isSome()) {
+      command.add("-ml");
+      command.add(Integer.toString(whispercppMaxLength.get()));
+    }
+    if (whispercppThreads.isSome()) {
+      command.add("-t");
+      command.add(Integer.toString(whispercppThreads.get()));
+    }
+    if (whispercppProcessors.isSome()) {
+      command.add("-p");
+      command.add(Integer.toString(whispercppProcessors.get()));
+    }
+    if (whispercppMaxContext.isSome()) {
+      command.add("-mc");
+      command.add(Integer.toString(whispercppMaxContext.get()));
+    }
+    if (whispercppSplitOnWord.isSome() && whispercppSplitOnWord.get()) {
+      command.add("-sow");
+    }
+    if (whispercppBestOf.isSome()) {
+      command.add("-bo");
+      command.add(Integer.toString(whispercppBestOf.get()));
+    }
+    if (whispercppWordThreshold.isSome()) {
+      command.add("-wt");
+      command.add(String.format("%f", whispercppWordThreshold.get()));
+    }
+    if (whispercppEntropyThreshold.isSome()) {
+      command.add("-et");
+      command.add(String.format("%f", whispercppEntropyThreshold.get()));
+    }
+    if (whispercppLogProbThreshold.isSome()) {
+      command.add("-lpt");
+      command.add(String.format("%f", whispercppLogProbThreshold.get()));
+    }
+    if (whispercppDiarization.isSome() && whispercppDiarization.get()) {
+      command.add("-di");
+    }
+    if (whispercppTinyDiarization.isSome() && whispercppTinyDiarization.get()) {
+      command.add("-tdrz");
+    }
+    if (whispercppNoFallback.isSome() && whispercppNoFallback.get()) {
+      command.add("-nf");
+    }
+
+    String subtitleLanguage;
+
+    // set language of the source audio if known
+    if (!language.isBlank()) {
+      logger.info("Using language {} from workflows", language);
+      command.add("--language");
+      command.add(language);
+    } else {
+      logger.debug("Auto-detecting language");
+      command.add("--language");
+      command.add("auto");
```

Comment:
I think this is superfluous since auto-detection is the default anyway. There is no need to pass on this parameter, is there?

Reply:
Actually no, the default in whispercpp is English, so it's needed for auto detection

Reply:
So my setup is defaulting to 
` --language eng`
which spits out 
```
whisper_lang_id: unknown language 'eng'
error: unknown language 'eng'
```

By default, if I did not set the language in the workflow I would get `eng`, rather than `english`, or `auto`.

Reply:
It seems the workflow defaults to using `eng` if the language code isn't given, eventhough the speechtotext-woh documentation says that language codes are the two latter codes defined in ISO 639-1. Since whispercpp only accepts either the two latter codes or the full lowercase name it cannot recognize `eng` and thus throws an error.